src/check_python_versions/sources/poetry_pyproject.py

```diff
@@ -38,11 +39,29 @@
 PYPROJECT_TOML = 'pyproject.toml'
 
 
+def _load_toml(filename: FileOrFilename):
+    table = {}
+    # tomlkit has two different API to load from file name or file object
+    if isinstance(filename, str):
+        with open_file(filename) as fp:
+            table = load(fp)
+    if isinstance(filename, TextIO):
+        table = load(filename)
```

Comment:
There's a `check_python_version.utils.open_file()`  context manager that could simplify this.